tfx_bsl/beam/run_inference.py

```diff
@@ -221,17 +223,129 @@ def update(self, elements: List[Union[tf.train.Example,
       self._inference_request_batch_byte_size.update(
           sum(element.ByteSize() for element in elements))
 
+  def __init__(self):
+    super(_BaseDoFn, self).__init__()
+    self._clock = None
+    self._metrics_collector = self._MetricsCollector()
+
+  def setup(self):
+    self._clock = _ClockFactory.make_clock()
+
+  def finish_bundle(self):
+    self._metrics_collector.update_metrics_with_cache()
+
+  def process(self, elements: list) -> Any:
+    batch_start_time = self._clock.get_current_time_in_microseconds()
+    result = self.run_inference(elements)
+    self._metrics_collector.update(
+        elements,
+        self._clock.get_current_time_in_microseconds() - batch_start_time)
+    return result
+
+  def run_inference(
+      self, elements: List[Union[tf.train.Example, tf.train.SequenceExample]]
+  ) -> list:
```

Comment:
Could this be a pytype type?
For saved_model_spec, this is Mapping[Text, np.ndarray]. If you need to extend it to a list, you can use Union[List,...]